modules/primo/includes/primo.search.inc

```diff
@@ -178,7 +178,23 @@ function primo_search_search(TingSearchRequest $ting_query) {
 
   // Handle raw query.
   if (!empty($raw_query = $ting_query->getRawQuery())) {
-    $queries['query'][] = htmlspecialchars($raw_query);
+    // The query starts with "[", so we query for comma separated ID's.
+    if (strpos($raw_query, '[') !== FALSE) {
+      $raw_ids = explode(',', trim($raw_query, '[]'));
+      $ids = [];
+      foreach ($raw_ids as $raw_id) {
+        $ids[] = $raw_id;
+      }
+      $queries['query'][] = 'rid,contains,' . implode(' OR ', $ids);
+    }
+    // The query starts with '"', so we search for the title.
+    else if (strpos($raw_query, '"') !== FALSE) {
+      $queries['query'][] = htmlspecialchars('swtitle,begins_with,' . trim($raw_query, '"'));
```

Comment:
We should have a general purpose quote function for primo - aaand I just realized that we're missing an implementation of search_quote for primo (https://github.com/rvk-utd/ding2/blob/37d0239f1ce8e54ecff292ad4deba8a5b866581e/modules/opensearch/includes/opensearch.search.inc#L740 ) so - how about adding a primo_search_search_quote() function in primo.search.inc and having it doing the escaping.

I'm not 100% we need the htmlspecialchars - but I guess it can't harm (?) - one thing we do need to look out for is commas - any commeas in the "value" part (3th argument to query) needs to be replaced with spaces.